### ISSUE TITLE
Can't share my position

### ISSUE THREAD

**Issue by atlochowski** [Karry/osmscout-sailfish#262]
OSMScout: 2.14

Steps to reproduce

1. Open menu
2.  Choose "Where am I?"
3.  Tap share icon

Result:  "Couldn't load page"

Logs:

```
lip 23 04:46:38 Sailfish [6992]: [D] onTap:474 - tap: 132x157 @ 52.15681514766271 21.048474638346555 (map center 52.156112085798526 21.050077592364794)
lip 23 04:46:45 Sailfish [6992]: [D] osmscout::PlaneMapRenderer::~PlaneMapRenderer:78 - ~PlaneMapRenderer
lip 23 04:46:45 Sailfish [6992]: [D] osmscout::TileLoaderThread::TileLoaderThread:32 - Overlay tile cache: osmscout::TileCache(0xb9119824)
lip 23 04:46:45 Sailfish [6992]: [D] osmscout::MapRenderer::~MapRenderer:74 - ~MapRenderer
lip 23 04:46:45 Sailfish [6992]: [D] osmscout::TiledMapRenderer::Initialize:95 - Initialize
lip 23 04:46:45 Sailfish [6992]: [D] IconProvider::requestPixmap:43 - Loading icon  "///usr/share/harbour-osmscout/pics/new-place.svg"
lip 23 04:46:45 Sailfish [6992]: [D] IconProvider::requestPixmap:43 - Loading icon  "///usr/share/harbour-osmscout/poi-icons/building.svg"
lip 23 04:46:45 Sailfish [6992]: [D] osmscout::LocationInfoModel::onLocationDescription:272 - Place  "52,15624 N 21,05045 E"  description:  3.58428  m  "E"  from  "Nugat 8, 02-776 Ursynów"
lip 23 04:46:49 Sailfish mce[1411]: modules/battery-udev.c: mcebat_update(): battery_level : 96 -> 97
lip 23 04:46:55 Sailfish dbus-daemon[5249]: dbus-daemon[5249]: [session uid=100000 pid=5249] Activating via systemd: service name='org.freedesktop.Tracker1' unit='tracker-store.service' requested by ':1.49' (uid=100000 pid=6794 comm="/usr/libexec/tracker-miner-fs")
lip 23 04:46:55 Sailfish systemd[5179]: Starting Tracker metadata database store and lookup manager...
lip 23 04:46:55 Sailfish [6992]: [W] unknown:1283 - file:///usr/lib/qt5/qml/Sailfish/Silica/PageStack.qml:1283:13: QML AnimatedLoader: (file:///usr/lib/qt5/qml/Sailfish/TransferEngine/SharePage.qml: File not found)
lip 23 04:46:55 Sailfish dbus-daemon[5249]: dbus-daemon[5249]: [session uid=100000 pid=5249] Successfully activated service 'org.freedesktop.Tracker1'
lip 23 04:46:55 Sailfish systemd[5179]: Started Tracker metadata database store and lookup manager.
```

```

[root@Sailfish TransferEngine]# ls /usr/lib/qt5/qml/Sailfish/TransferEngine/
ShareFilePreview.qml        SharePostPreview.qml        TransfersPage.qml           libsilicatransferengine.so  qmldir
```

**Comment by Karry** [comment]
It is strange, I have this page on my Xperia 10 II (SFOS 4.1.0.24) at `/usr/lib64/qt5/qml/Sailfish/TransferEngine/SharePage.qml`, it is part of `declarative-transferengine-qt5` package:
```
[root@Xperia ~]# rpm -qf /usr/lib64/qt5/qml/Sailfish/TransferEngine/SharePage.qml
declarative-transferengine-qt5-0.3.21-1.14.1.jolla.aarch64
```

...and on my old Jolla 1 (SFOS 3.4.0.24):
```
[root@Sailfish ~]# rpm -qf /usr/lib/qt5/qml/Sailfish/TransferEngine/SharePage.qml
declarative-transferengine-qt5-0.3.17.1-1.13.1.jolla.armv7hl
```

**Comment by atlochowski** [comment]
I have sent you message on forum

**Comment by Karry** [comment]
I dont think that [this permission issue](https://forum.sailfishos.org/t/sharing-via-email-fails-from-ordinary-application/7411) is related to your problem. It is sharing from other applications like Notes working for you?

**Comment by Karry** [comment]
Just to clarify - system API for sharing was redesigned in SFOS 4.2.*. I believe that this change was needed for handling sharing permission... Old API that was using `/usr/lib[64]/qt5/qml/Sailfish/TransferEngine/SharePage.qml` does't work anymore. 

As SFOS 4.2 was released yesterday for early access subscribers, I can dig deeper. There is new `ShareAction` Silica component (looking to Notes QML sources). It is not documented yet... Lets see if it will be working.

**Comment by Karry** [comment]
This component is not allowed in Harbour yet :-( Package validation fails with:
```
ERROR [/usr/share/harbour-osmscout/qml/pages/PlaceDetail.qml] Import 'Sailfish.Share 1.0' is not allowed
```

It is not allowed even in upstream validator. So I created ticket for that: https://github.com/sailfishos/sdk-harbour-rpmvalidator/issues/135

**Comment by Karry** [comment]
Should be fixed in 2.16
